### PR TITLE
fix: DREAMTEM-3-Basic UI kit components

### PR DESCRIPTION
**Pull request by PolliShell**
## Describe your changes

- I renamed the components and fixed errors from the past pr.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://zenbitdreamteam.atlassian.net/browse/DREAMTEAM-3?atlOrigin=eyJpIjoiN2Y2ZDZlMDAyMzMyNGM2YThkN2U3NTIzYzI0OTk1NTAiLCJwIjoiaiJ9)

### **General**

- [x] Types for input and output params
- [x ] No `any`, should be also added to eslint rules
- [ ] try/catch for any async function
- [x] no magic numbers (describe it in constants)
- [x] compare only with constants not with strings (instead of user === ‘adminʼ, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no console.log in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL should be in constant file
- [x] constants and function names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyyˮ), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _ʼ, const EACH_DAY_15h_15min = ‘15 3 _ \* \*ʼ

### Frontend

- [x] split component and logic (move logic to custom hook file)
- [x] colors, font size, and font name should be in the theme or in the constants
- [x] no text in the components, use i18n approach, even if you have only one language for now
- [x] inline styles prohibited
- [x] import should be absolute. instead of ../../../components/myComponent should be components/myComponent

### CODE REVIEW

File: src/utils/stringUtils.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const getInitials = (name: string): string => {
```
Comment (lyudad) [inline src/utils/stringUtils.ts:1]: need example of input and output params, so everyone understand does it work

File: src/utils/colorUtils.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const hexToRgb = (hex: string): string => {
```
Comment (lyudad) [inline src/utils/colorUtils.ts:1]: need example of input and output params, so everyone understand does it work

Comment (lyudad) [reply src/utils/colorUtils.ts:1]: something similar https://stackoverflow.com/a/14419647 but with example

**Comment by PolliShell** [comment]
Вы апрувите это или ещё нет?

чт, 7 нояб. 2024 г., 23:38 Liudmyla ***@***.***>:

**Pull request merged**